### DISCUSSION TITLE
Does Apache Superset support window functions like Rank, Lead, Lag etc

### DISCUSSION THREAD

**Discussion by pthirumalai** [apache/superset#23049] [General]
Hi

Does Apache Superset support window functions or analytical functions like Rank, Lead, Lag etc.

Thanks
Thirumalai

**Comment by sfirke** [comment]
Superset is a thin layer over your data warehouse.  You can create virtual datasets that are SQL queries on top of your data warehouse, so I would think you could create a virtual dataset with window functions - but I build my data sources as Views outside of Superset so can't say firsthand.

**Comment by villebro** [comment]
@pthirumalai Superset is fully SQL based, meaning any syntax your underlying database supports, you can also use in Superset.
